zebra/redistribute.c

```diff
@@ -547,8 +547,10 @@ int zebra_add_import_table_entry(struct route_node *rn, struct route_entry *re,
 			break;
 	}
 
-	if (same)
+	if (same) {
+		UNSET_FLAG(same->flags, ZEBRA_FLAG_SELECTED);
```

Comment:
is there a reason to *not* also unset the flag before the earlier call to zebra_del_import_table_entry (@line 534)?

Reply:
I can verify that moving it to inside the loop also resolves the issue but please explain why that would be better than what I've done?

Reply:
As expected, that also works to resolve the problem. I would still like to understand how this is better than what I did for my own education.

Reply:
Sorry, I completely misunderstood your comment.  Ignore me until after I revisit.  Sorry for the noise.

Reply:
I agree with your comment and will add the unset at that location as well.

Reply:
changes uploaded